source/string.c

```diff
@@ -15,16 +15,15 @@
 #include <aws/common/string.h>
 
 struct aws_string *aws_string_new_from_c_str(struct aws_allocator *allocator, const char *c_str) {
-    if (c_str == NULL || allocator == NULL) {
-        return NULL;
-    }
+    AWS_FATAL_ASSERT(allocator && c_str);
```

Comment:
Should this be a `PRECONDITION`?

Reply:
@ColdenCullen They should be `AWS_FATAL_PRECONDITION` to make sure we check them in prod. However, since the macro isn't available yet, I chose to use `AWS_FATAL_ASSERT` so it'll be easier to identify it and replace it when https://github.com/awslabs/aws-c-common/pull/430 get merged to master. What do you think?

Reply:
I'm not convinced they're needed in release builds, something like a null parameter should be catchable in debug before you ever hit a release build. What do you think @justinboswell ?

Reply:
```AWS_PRECONDITION``` seems sufficient to me. If we're proving that the paths in are clean, the runtime check seems excessive.

Reply:
:+1:

Reply:
Done.